internal/pkg/build/sources/conveyorPacker_oci.go

```diff
@@ -110,35 +112,32 @@ exec "$@"
 
 // OCIConveyorPacker holds stuff that needs to be packed into the bundle
 type OCIConveyorPacker struct {
-	srcRef    types.ImageReference
-	b         *sytypes.Bundle
-	imgConfig imgspecv1.ImageConfig
-	sysCtx    *types.SystemContext
+	srcRef           types.ImageReference
+	b                *sytypes.Bundle
+	imgConfig        imgspecv1.ImageConfig
+	transportOptions *ocitransport.TransportOptions
 }
 
 // Get downloads container information from the specified source
 func (cp *OCIConveyorPacker) Get(ctx context.Context, b *sytypes.Bundle) (err error) {
 	cp.b = b
 
-	// DockerInsecureSkipTLSVerify is set only if --no-https is specified to honor
-	// configuration from /etc/containers/registries.conf because DockerInsecureSkipTLSVerify
-	// can have three possible values true/false and undefined, so we left it as undefined instead
-	// of forcing it to false in order to delegate decision to /etc/containers/registries.conf:
-	// https://github.com/sylabs/singularity/issues/5172
-	cp.sysCtx = &types.SystemContext{
-		OCIInsecureSkipTLSVerify: cp.b.Opts.NoHTTPS,
-		DockerAuthConfig:         cp.b.Opts.DockerAuthConfig,
-		DockerDaemonHost:         cp.b.Opts.DockerDaemonHost,
-		AuthFilePath:             ociauth.ChooseAuthFile(cp.b.Opts.DockerAuthFile),
-		DockerRegistryUserAgent:  useragent.Value(),
-		BigFilesTemporaryDir:     b.TmpDir,
-		OSChoice:                 cp.b.Opts.Platform.OS,
-		ArchitectureChoice:       cp.b.Opts.Platform.Architecture,
-		VariantChoice:            cp.b.Opts.Platform.Variant,
-	}
-
-	if cp.b.Opts.NoHTTPS {
-		cp.sysCtx.DockerInsecureSkipTLSVerify = types.NewOptionalBool(true)
+	cp.transportOptions = &ocitransport.TransportOptions{
+		Insecure:         cp.b.Opts.NoHTTPS,
+		DockerDaemonHost: cp.b.Opts.DockerDaemonHost,
+		AuthConfig:       cp.b.Opts.OCIAuthConfig,
+		AuthFilePath:     ociauth.ChooseAuthFile(cp.b.Opts.DockerAuthFile),
+		UserAgent:        useragent.Value(),
+		TmpDir:           b.TmpDir,
+		Platform:         cp.b.Opts.Platform,
+	}
+
+	if cp.b.Opts.OCIAuthConfig == nil && cp.b.Opts.DockerAuthConfig != nil {
```

Comment:
I'm *guessing* that the reason `cp.b.Opts` now has both an `OCIAuthConfig` field and a `DockerAuthConfig` (as opposed to unifying them into a single `OCIAuthConfig`, as has been done elsewhere) is that `pkg/build/types` is under the `pkg/` toplevel. Can you confirm?

Reply:
Yes - we have to keep `DockerAuthConfig` there, and working until v5.